hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/domain/ContractResultServiceImplIntegrationTest.java

```diff
@@ -346,6 +346,84 @@ void processContractActionMissingCaller() {
         assertThat(entityRepository.count()).isZero();
     }
 
+    @Test
+    void processGasConsumedCalculationContractCall() {
+        // Given RecordItem with 3 contract actions
+        final var recordItem = recordItemBuilder.contractCall().build();
+
+        final var contractActionRecord = TransactionSidecarRecord.newBuilder()
+                .setActions(ContractActions.newBuilder()
+                        .addContractActions(
+                                contractAction(3001L, CallOperationType.OP_CALL, ContractActionType.CALL, 3002L))
+                        .addContractActions(
+                                contractAction(3003L, CallOperationType.OP_CREATE, ContractActionType.CREATE, 3004L))
+                        .addContractActions(contractAction(
+                                3004L, CallOperationType.OP_DELEGATECALL, ContractActionType.CALL, 3005L)))
+                .build();
+
+        recordItem.setSidecarRecords(List.of(contractActionRecord));
+
+        // When
+        process(recordItem);
+
+        // Then
+        final var contractResult = contractResultRepository
+                .findById(recordItem.getConsensusTimestamp())
+                .orElse(null);
+        assertThat(contractResult).isNotNull();
+        assertThat(contractResult.getGasConsumed()).isNotNull();
+        assertThat(contractResult.getGasConsumed()).isEqualTo(21150L);
```

Comment:
Take advantage of assertj's chaining. Same comment for other tests.

```suggestion
        assertThat(contractResultRepository.findById(recordItem.getConsensusTimestamp())
            .isNotNull()
            .extracting(ContractResult::getGasConsumed)
            .isNotNull()
            .isEqualTo(21150L);
```